src/Google/Ads/GoogleAds/Util/FieldMasks.php

```diff
@@ -163,6 +163,7 @@ private static function buildPaths(
                     // beginning.
                     (!is_null($original) && method_exists($original, $hasser)
                         && $original->$hasser() != $modified->$hasser())
+                    || (is_null($original) && !$modifiedValue)
```

Comment:
I'm a bit confused by this.
What are we trying to check here?
Looks like if original is null and the modified value is false then this will be marked as a value changed.
What about the case when the modified value is true?

Reply:
This is to alleviate the fact that when the original is non-existent and the modified value is  a `false` boolean then the `$originalValue != $modifiedValue` condition returns `false`.

This is due to some inconsistency in PHP's `==` operator:

```php
print (null == false);

//> 1
```

Reply:
Then, why don't we just modify the code below to

```php
|| $originalValue !== $modifiedValue;
```

instead?
That should work for both `true` and `false` cases, without the need to add more conditions.

Reply:
Yep, please do.

If necessary, we can split comparison when the field type is MESSAGE and when it's not, for clearer logic.

Reply:
The confusion came from how the unit tests were written: the expected values were computed at runtime by the library instead of hard-coded and used an assert that was always returning `true` even if untrue. I upgraded the tests to avoid these issues.

We now have one clear test case that reproduces the exact problem with both implementations of protobuf.